c7n/resources/ssm.py

```diff
@@ -16,21 +16,54 @@
 
 from c7n.query import QueryResourceManager
 from c7n.manager import resources
-from c7n.utils import get_retry
-
+from c7n.utils import get_retry, local_session, type_schema
+from c7n.actions import Action
 
 @resources.register('ssm-parameter')
 class SSMParameter(QueryResourceManager):
-
     class resource_type(object):
-        service = 'ssm'
-        enum_spec = ('describe_parameters', 'Parameters', None)
-        name = "Name"
-        id = "Name"
-        filter_name = None
-        dimension = None
-        universal_taggable = True
+      service = 'ssm'
+      enum_spec = ('describe_parameters', 'Parameters', None)
+      name = "Name"
+      id = "Name"
+      filter_name = None
+      dimension = None
+      universal_taggable = True
 
     retry = staticmethod(get_retry(('Throttled',)))
     permissions = ('ssm:GetParameters',
                    'ssm:DescribeParameters')
+
+@resources.register('ssm-managed-instances')
+class ManagedInstance(QueryResourceManager):
+    class resource_type(object):
+        service = 'ssm'
+        enum_spec = ('describe_instance_information', 'InstanceInformationList', None)
+        id = 'InstanceId'
+        name = 'Name'
+        date = 'RegistrationDate'
+        dimension = None
+        filter_name = None
```

Comment:
this needs manual filter construction for use in serverless policies, ie. the api supports server side filtering but in a form thats not amenable to extant query support. We typically use a custom DescribeSource here, looks likes the same for all of the ssm resources, and we should just add query.py support for the form here, that can be separate from this pr though.